### PR TITLE
Rootfs image name setup

### PR DESCRIPTION
**Pull request by jbrun3t**
Here is a proposition to JSM request.
Sent here for us to discuss. If this as to go through Gerrit, please don't merge it

### CODE REVIEW

File: templates/base/agl-base-defaults.jinja2 [hunk 1]
```diff
@@ -21,13 +21,13 @@
...
 {%- set boot_commands = "nbd" %}
-{%- set nbdroot = nbdroot|default("core-image-minimal-" + yocto_machine + ".ext4.xz") %}
+{%- set nbdroot = rfs_image|default("core-image-minimal-" + yocto_machine + ".ext4.xz") %}
 {%- set nbdroot_url = nbdroot_url|default(baseurl(nbdroot)) %}
 {%- set nbdroot_compression = nbdroot_compression|default(nbdroot|get_extension) %}
 {%- set nbdinitrd = nbdinitrd|default("initramfs-netboot-image-" + yocto_machine +".ext4.gz") %}
 {%- set nbdinitrd_url = nbdinitrd_url|default(baseurl(nbdinitrd)) %}
 {%- elif rootfs_type == 'ramdisk' %}
-{%- set initrd = initrd|default("initramfs-boot-image-" + yocto_machine +".gz") %}
+{%- set initrd = rfs_image|default("agl-demo-platform-" + yocto_machine + ".ext4.xz") %}
```
Comment (khilman) [inline templates/base/agl-base-defaults.jinja2]: I don't get why you're changing the initrd from initramfs-* to agl-demo-platform-*.  Seems unrelated to this patch?

Comment (jbrun3t) [reply templates/base/agl-base-defaults.jinja2]: In the first version of releng I was only trying to mimic the jobs posted on agl's lava as they are ... no care for consistency.

No that we can specify the image name, I thought it would be important that the default value is consistent between ramdisk and nbd

Do you think I should keep the previous value ?

File: utils/create-jobs.py [hunk 2]
```diff
@@ -32,14 +32,28 @@ def parse_cmdline(machines, tests, rfs_types):
                         help='job id for link creation: URLBASE/JOB_ID')
     parser.add_argument('-i', '--jobidx', dest='job_index',  action='store',
                         help='job index for link creation: URLBASE/JOB_ID/JOB_INDEX', default='1')
+    parser.add_argument('--img-name', dest='img_name',  action='store',
+                        help="img base name (such as agl-demo-platform) - require img_ext")
+    parser.add_argument('--img-ext', dest='img_ext',  action='store',
+                        help="img extension (such as ext4.xz) - require img_name")
```
Comment (loysollivier) [inline utils/create-jobs.py:38]: Question: maybe there's a way to add mutual requirement.
Or if not maybe we should pass it together: --image agl-demo-platform .ext4.xz 

This can be an improvement

File: utils/create-jobs.py [hunk 3]
```diff
@@ -32,14 +32,28 @@ def parse_cmdline(machines, tests, rfs_types):
...
 
-    return parser.parse_args()
+    args = parser.parse_args()
+
+    if (not args.img_name) != (not args.img_ext):
```
Comment (loysollivier) [inline utils/create-jobs.py:42]: Ok so this answers my comment 4 lines above.

File: utils/create-jobs.py [hunk 4]
```diff
@@ -32,14 +32,28 @@ def parse_cmdline(machines, tests, rfs_types):
...
+    return args
 
 
 def main():
     ajt = agljobtemplate.Agljobtemplate('templates')
     args = parse_cmdline(ajt.machines, ajt.tests, ajt.rfs_types)
 
+    if args.img_name:
+        img = args.img_name + "-" + args.machine + "." + args.img_ext
+    else:
```
Comment (loysollivier) [inline utils/create-jobs.py]: nitpicking-on: I would declare `img = None` at the start of the function and just do the if here nitpicking-off

It totally works like that

**Review by loysollivier — APPROVED** [review]
Looks good to me.

File: templates/base/agl-base-defaults.jinja2 (hunk 1 above)
Comment (khilman) [reply templates/base/agl-base-defaults.jinja2]: Yes, the defafult initrd image should be initramfs-boot-image.  It's only on QEMU that it needs to be overridden, so I would leave it that way.  I find it a bit confusing in this patch.

**Comment by jbrun3t** [comment]
Comments addressed. Thx Guys

**Comment by khilman** [comment]
LGTM, can you submit to gerrit?
